pgmpy/models/MarkovChain.py

```diff
@@ -367,6 +368,50 @@ def generate_sample(self, start_state=None, size=1):
                 self.state[j] = State(var, next_st)
             yield self.state[:]
 
+    def check_stationarity(self, tolerance=0.2, sample=None):
+        """
+        Checks if the given markov model is stationary and checks the steady state
+        probablity values for the state are consistent.
+
+        Parameters:
+        -----------
+        tolerance: float
+            reprsents the diff between actual steady state value and the computed value
+        sample: [State(i,j)]
+            represents the list of state which the markov chain has sampled
+
+        Return Type:
+        ------------
+        None
+
+        Examples:
+        ---------
+        >>> from pgmpy.models.MarkovChain import MarkovChain
+        >>> from pgmpy.factors.discrete import State
+        >>> model = MarkovChain()
+        >>> model.add_variables_from(['intel', 'diff'], [3, 2])
+        >>> intel_tm = {0: {0: 0.2, 1: 0.4, 2:0.4}, 1: {0: 0, 1: 0.5, 2: 0.5}, 2: {0: 0.3, 1: 0.3, 2: 0.4}}
+        >>> model.add_transition_model('intel', intel_tm)
+        >>> diff_tm = {0: {0: 0.5, 1: 0.5}, 1: {0: 0.25, 1:0.75}}
+        >>> model.add_transition_model('diff', diff_tm)
+        >>> model.check_stationarity()
+        True
+        """
+        keys = self.transition_models.keys()
+        for k in keys:
+            transition_mat = DataFrame(self.transition_models[k]).as_matrix()
+            S, U = eig(transition_mat.T)
+            stationary = np.array(U[:, np.where(np.abs(S - 1.) < 1e-8)[0][0]].flat)
+            stationary = (stationary / np.sum(stationary)).real
+            probabilites = []
+            window_size = 10000 if sample == None else len(sample) 
```

Comment:
@ankurankan Yes i know that burn-in period depends upon the start state and transition probabilities that's why I was suggesting not to use some large value. This might not sound good but i was thinking using a random value out of a set of most likely values (like choosing a random value out of (100, 1000, 10000) or something like that.